std/container/array.d

```diff
@@ -355,8 +354,11 @@ if (!is(Unqual!T == bool))
                  * than realloc.
                  */
                 immutable oldLength = length;
-                auto newPayload =
-                    enforce(cast(T*) malloc(sz))[0 .. oldLength];
+
+                auto newPayloadPtr = cast(T*) malloc(sz);
+                assert(newPayloadPtr, "std.container.Array.reserve failed to allocate memory");
```

Comment:
Let's do this: ` newPayloadPtr || assert(false, "std.container.Array.reserve failed to allocate memory");`